vamb/taxonomy.py

```diff
@@ -176,6 +177,9 @@ def __init__(
         is_canonical: bool,
     ):
         if len(taxonomies) != len(metadata.identifiers):
+            logger.error(
+                f"Taxonomies length: {len(taxonomies)}, Identifiers length: {len(metadata.identifiers)}"
+            )
```

Comment:
I believe all exceptions are already caught and logged by the logger. Instead, improve the `ValueError`